linera-views/src/memory.rs

```diff
@@ -130,12 +108,135 @@ impl KeyValueStoreClient for MemoryClient {
     }
 }
 
+/// Implementation of the MemoryClient from the existing one.
+#[derive(Clone)]
+pub struct MemoryClient {
+    client: KeyValueStoreClientBigValue<MemoryClientInternal>,
+}
+
+#[async_trait]
+impl KeyValueStoreClient for MemoryClient {
+    const MAX_CONNECTIONS: usize = MemoryClientInternal::MAX_CONNECTIONS;
+    const MAX_VALUE_SIZE: usize = MemoryClientInternal::MAX_VALUE_SIZE;
+    type Error = MemoryContextError;
+    type Keys = Vec<Vec<u8>>;
+    type KeyValues = Vec<(Vec<u8>, Vec<u8>)>;
+
+    async fn read_key_bytes(&self, key: &[u8]) -> Result<Option<Vec<u8>>, MemoryContextError> {
+        self.client.read_key_bytes(key).await
+    }
+
+    async fn read_multi_key_bytes(
+        &self,
+        keys: Vec<Vec<u8>>,
+    ) -> Result<Vec<Option<Vec<u8>>>, MemoryContextError> {
+        self.client.read_multi_key_bytes(keys).await
+    }
+
+    async fn find_keys_by_prefix(
+        &self,
+        key_prefix: &[u8],
+    ) -> Result<Self::Keys, MemoryContextError> {
+        self.client.find_keys_by_prefix(key_prefix).await
+    }
+
+    async fn find_key_values_by_prefix(
+        &self,
+        key_prefix: &[u8],
+    ) -> Result<Self::KeyValues, MemoryContextError> {
+        self.client.find_key_values_by_prefix(key_prefix).await
+    }
+
+    async fn write_batch(&self, batch: Batch, base_key: &[u8]) -> Result<(), MemoryContextError> {
+        self.client.write_batch(batch, base_key).await
+    }
+
+    async fn clear_journal(&self, base_key: &[u8]) -> Result<(), Self::Error> {
+        self.client.clear_journal(base_key).await
+    }
+}
+
+impl MemoryClient {
+    /// Create a MemoryClient from the guard
+    pub fn new(guard: MutexGuardArc<MemoryStoreMap>) -> Self {
+        let client = Arc::new(RwLock::new(guard));
+        MemoryClient {
+            client: KeyValueStoreClientBigValue::new(client),
+        }
+    }
+}
+
+/// An implementation of [`crate::common::Context`] that stores all values in memory.
+pub type MemoryContext<E> = ContextFromDb<E, MemoryClient>;
+
+impl<E> MemoryContext<E> {
+    /// Creates a [`MemoryContext`].
+    pub fn new(guard: MutexGuardArc<MemoryStoreMap>, extra: E) -> Self {
+        let db = MemoryClient::new(guard);
+        let base_key = Vec::new();
+        Self {
+            db,
+            base_key,
+            extra,
+        }
+    }
+}
+
+/// Provides a `MemoryContext<()>` that can be used for tests.
+/// It is not named create_memory_test_context because it is massively
+/// used
+pub fn create_test_context() -> MemoryContext<()> {
+    let state = Arc::new(Mutex::new(BTreeMap::new()));
+    let guard = state
+        .try_lock_arc()
+        .expect("We should acquire the lock just after creating the object");
+    MemoryContext::new(guard, ())
+}
+
+/// create a test memory client for working.
```

Comment:
What does "for working" mean? Not test-only?

```suggestion
/// Creates a test memory client for working.
```